cdap-security/src/main/java/io/cdap/cdap/security/tools/AccessTokenGeneratorService.java

```diff
@@ -0,0 +1,228 @@
+      } catch (IllegalArgumentException e) {
+        throw new BadRequestException(String.format("Invalid identifierType. Valid values: %s",
+                                                    Joiner.on(',').join(UserIdentity.IdentifierType.values())));
+      }
+      long expirationTime = cConf.getLong(Constants.Security.TOKEN_EXPIRATION);
+      long issueTime = System.currentTimeMillis();
+      long expireTime = issueTime + expirationTime;
+      UserIdentity userIdentity = new UserIdentity(username, type,
+                                                   Collections.emptyList(),
+                                                   issueTime, expireTime);
+      AccessToken token = tokenManager.signIdentifier(userIdentity);
+      String encodedToken = Base64.getEncoder().encodeToString(tokenCodec.encode(token));
+      responder.sendString(HttpResponseStatus.OK, encodedToken);
+    }
+  }
+
+  @VisibleForTesting
+  static Injector createInjector(CConfiguration cConf) {
+    return Guice.createInjector(
+      new ConfigModule(cConf),
+      new IOModule(),
+      new AbstractModule() {
+        @Override
+        protected void configure() {
+          // We don't do anything with metrics, so just bind to the no-op implementation
+          bind(MetricsCollectionService.class).to(NoOpMetricsCollectionService.class);
+        }
+      },
+      CoreSecurityRuntimeModule.getDistributedModule(cConf));
+  }
+
+  public static void main(String[] args)throws Exception {
+    Option portOption = new Option("p", "port", true, "Server port.");
+    portOption.setType(Integer.class);
+    Option workerPoolSizeOption = new Option("w", "workerPoolSize", true,
+                                             "Size of worker thread pool.");
+    workerPoolSizeOption.setType(Integer.class);
+    Option execPoolSizeOption = new Option("e", "execPoolSize", true,
+                                           "Size of executor thread pool.");
+    execPoolSizeOption.setType(Integer.class);
+    Option tokenExpirationOption = new Option("t", "tokenExpiration", true,
+                                           "Access token expiration time in milliseconds.");
+    tokenExpirationOption.setType(Integer.class);
+
+    Options options = new Options()
+      .addOption(new Option("h", "help", false, "Print this usage message."))
+      .addOption(new Option("a", "address", true, "Server IP address."))
+      .addOption(portOption)
+      .addOption(workerPoolSizeOption)
+      .addOption(execPoolSizeOption)
+      .addOption(tokenExpirationOption);
+
+    CommandLineParser parser = new BasicParser();
+    CommandLine commandLine = parser.parse(options, args);
+
+    CConfiguration cConf = CConfiguration.create();
+    cConf.set(ADDRESS_CONFIG, ADDRESS_DEFAULT);
+    if (commandLine.hasOption("a")) {
+      cConf.set(ADDRESS_CONFIG, commandLine.getOptionValue("a"));
+    }
+    cConf.setInt(PORT_CONFIG, PORT_DEFAULT);
```

Comment:
You don't need this. You can do a get with default in the constructor instead.

```
cConf.getInt(PORT_CONFIG, PORT_DEFAULT);
```

Same for all other configs

Reply:
fixed